cleanlab_studio/utils/data_enrichment/enrich.py

```diff
@@ -94,41 +74,78 @@ def enrich_data(
+            )
 
-    if constrain_outputs:
-        df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
-            lambda x: get_constrain_outputs_match(
-                x, constrain_outputs, disable_warnings=disable_warnings
+        if constrain_outputs:
+            df[f"{new_column_name}"] = df[f"{new_column_name}"].apply(
+                lambda x: get_constrain_outputs_match(x, constrain_outputs)
             )
-        )
 
-    return df[
-        [
-            f"{new_column_name}",
-            f"{column_name_prefix}trustworthiness",
-            f"{column_name_prefix}log",
+        enriched_df = df[
+            [
+                f"{new_column_name}",
+                f"{column_name_prefix}trustworthiness",
+                f"{column_name_prefix}log",
+            ]
         ]
-    ]
+
+    # Simulate the response structure of the run() method
+    job_info = {
+        "job_id": "run_online",
+        "status": "SUCCEEDED",
+        "num_rows": len(enriched_df),
+        "processed_rows": len(enriched_df),
+        "average_trustworthiness_score": enriched_df[f"{column_name_prefix}trustworthiness"].mean(),
+        "results": enriched_df,
+    }
+
+    return job_info
+
+
+def _validate_enrichment_options(options: EnrichmentOptions) -> None:
+    required_keys = ["prompt"]
+    for key in required_keys:
+        if key not in options or options[key] is None:
+            raise ValueError(f"'{key}' is required in the options.")
+
+    # Validate types and values
+    if not isinstance(options["prompt"], str):
+        raise TypeError("'prompt' must be a string.")
+
+    if "constrain_outputs" in options and options["constrain_outputs"] is not None:
+        if not isinstance(options["constrain_outputs"], list):
+            raise TypeError("'constrain_outputs' must be a list if provided.")
+
+    if "optimize_prompt" in options and options["optimize_prompt"] is not None:
+        if not isinstance(options["optimize_prompt"], bool):
+            raise TypeError("'optimize_prompt' must be a boolean if provided.")
+
+    if "quality_preset" in options and options["quality_preset"] is not None:
+        if not isinstance(options["quality_preset"], str):
+            raise TypeError("'quality_preset' must be a string if provided.")
+
+    if "regex" in options and options["regex"] is not None:
+        regex = options["regex"]
+        if not isinstance(regex, (str, tuple, list)):
+            raise TypeError("'regex' must be a string, tuple, or list of tuples.")
+        if isinstance(regex, list) and not all(isinstance(item, tuple) for item in regex):
+            raise TypeError("All items in 'regex' list must be tuples.")
 
 
 def process_regex(
-    column_data: Union[pd.Series, List[str]],
-    regex: Union[str, Replacement, List[Replacement]],
-) -> Union[pd.Series, List[str]]:
+    column_data: Union["pd.Series", List[str]],
+    regex: Union[str, Tuple[str, str], List[Tuple[str, str]]],
```

Comment:
[nit] `Replacement` is a type alias for the `Tuple[str, str]` type (ref [here](https://github.com/cleanlab/cleanlab-studio/blob/refactor-dynamic-static-enrichment-apis/cleanlab_studio/internal/enrichment_utils.py#L12)), not entirely sure why you made this change?